compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -516,12 +599,43 @@ abstract class GenJSCode extends plugins.PluginComponent
         if !f.isMethod && f.isTerm && !f.isModule
       } yield {
         implicit val pos = f.pos
+
         val mutable =
           suspectFieldMutable(f) || unexpectedMutatedFields.contains(f)
-        js.FieldDef(encodeFieldSym(f), toIRType(f.tpe), mutable)
+        val name =
+          if (jsInterop.isExposed(f)) js.StringLiteral(jsNameOf(f))
+          else encodeFieldSym(f)
+
+        val irTpe = {
+          if (f.tpe.typeSymbol != CharClass || !isScalaJSDefinedJSClass(cd.symbol))
```

Comment:
You are using `cd.symbol` here, above you use `currentClassSym`. Maybe have a `val` for the class symbol and assert the two are the same?